issuehunter_observer.go

```diff
@@ -1,67 +1,101 @@
 )
 
-func CreateFilter(ethRpcClient *ethrpc.EthRPC, contractAddress string) string {
-	log.Printf("creating a new event filter for the IssueHunter contract deployed at address %v", contractAddress)
-	var filter = map[string]interface{}{"address": contractAddress}
-	filterId, err := ethRpcClient.EthNewFilter(filter)
+func createFilter(ethRpcClient *ethrpc.EthRPC, contractAddress string, filter map[string]interface{}) string {
+	filterID, err := ethRpcClient.EthNewFilter(filter)
 	if err != nil {
 		log.Fatal(err)
 	}
-	log.Printf("new event filter for the IssueHunter contract deployed at address %v: %v", contractAddress, filterId)
-	return filterId
+	return filterID
+}
+
+func CreateFilter(ethRpcClient *ethrpc.EthRPC, contractAddress string) string {
+	log.Printf("creating a new event filter for the IssueHunter contract deployed at address %v", contractAddress)
+	var filter = map[string]interface{}{"address": contractAddress}
+	filterID := createFilter(ethRpcClient, contractAddress, filter)
+	log.Printf("new event filter for the IssueHunter contract deployed at address %v: %v", contractAddress, filterID)
+	return filterID
 }
 
-func GetEvents(ethRpcClient *ethrpc.EthRPC, filterId string) []interface{} {
-	log.Printf("getting changes for filter %v", filterId)
-	logs, err := ethRpcClient.EthGetFilterChanges(filterId)
+func GetEvents(ethRpcClient *ethrpc.EthRPC, filterID string) []interface{} {
+	log.Printf("getting changes for filter %v", filterID)
+	logs, err := ethRpcClient.EthGetFilterChanges(filterID)
 	if err != nil {
 		log.Fatal(err)
 	}
 	jsonLogs, _ := json.MarshalIndent(logs, "", "  ")
-	log.Printf("got changes for filter %v: %v", filterId, string(jsonLogs))
+	log.Printf("got changes for filter %v: %v", filterID, string(jsonLogs))
 	return logs
 }
 
-
-func UninstallFilter(ethRpcClient *ethrpc.EthRPC, filterId string) bool {
-	log.Printf("uninstalling filter %v", filterId)
-	uninstall, err := ethRpcClient.EthUninstallFilter(filterId)
+func UninstallFilter(ethRpcClient *ethrpc.EthRPC, filterID string) bool {
+	log.Printf("uninstalling filter %v", filterID)
+	uninstall, err := ethRpcClient.EthUninstallFilter(filterID)
 	if err != nil {
 		log.Fatal(err)
 	}
 	if !uninstall {
-		log.Fatalf("cannot uninstall filter %v", filterId)
+		log.Fatalf("cannot uninstall filter %v", filterID)
 	}
 
-	log.Printf("filter %v uninstalled", filterId)
+	log.Printf("filter %v uninstalled", filterID)
 	return uninstall
 }
 
+func CreateResolutionProposedFilter(ethRpcClient *ethrpc.EthRPC, contractAddress string) string {
+	log.Printf("creating a new filter on ResolutionProposed event for the IssueHunter contract deployed at address %v", contractAddress)
+	// > web3.sha3('ResolutionProposed(bytes32,address,bytes32)')
+	// "0x3b2c9742ac31922699b361e5e2c3d23b7762a77db8d2ae4b5d8004c904e9a4b7"
+	var filter = map[string]interface{}{
+		"address": contractAddress,
+		"topics":  []interface{}{"0x3b2c9742ac31922699b361e5e2c3d23b7762a77db8d2ae4b5d8004c904e9a4b7"},
```

Comment:
Not required, but we could compute this by using https://godoc.org/golang.org/x/crypto/sha3.